lib/suse/toolkit/renderer.rb

```diff
@@ -7,11 +7,21 @@ def initialize(*arguments)
         super
       end
 
-      def render(filename, locals: binding)
-        bind = locals.is_a?(Binding) ? locals : OpenStruct.new(locals).instance_eval { binding }
-        @templates[filename] ||= ERB.new File.read(File.expand_path("../../connect/templates/#{filename}.erb", __FILE__)), 0, '-<>'
+      def render(filename, locals = {})
+        bind = binding
+        path = "../../connect/templates/#{filename}.erb"
+
+        locals.each_pair do |key, value|
+          bind.local_variable_set(key, value)
+        end
```

Comment:
Why the change to the bindings logic?

Reply:
I want that the renderer class methods are within the context (aka render and indent method). The old implementation does work for the first call where the binding from the default parameter is used. But in case when locals are supplied it evaluates the binding within the context of the openstruct which works fine but does not export the methods of renderer, resulting in calling 'render' or 'indent' is not known every render call except to the first one.

Reply:
Ok got it!